src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -648,7 +644,7 @@ private static Details getDetailsForOthers(DetailAST ast) {
             final int tokenType = ast.getType();
             if (isTokenWithNoChildSlist(tokenType)) {
                 final DetailAST child = ast.getLastChild();
-                lcurly = child.getFirstChild();
+                lcurly = child;
```

Comment:
Tokens that enter here can only be in `TOKENS_WITH_NO_CHILD_SLIST` which is `TokenTypes.CLASS_DEF, TokenTypes.ENUM_DEF, TokenTypes.ANNOTATION_DEF, TokenTypes.INTERFACE_DEF, TokenTypes.RECORD_DEF`.

`child` is always the `OBJBLOCK` of these tokens since they are all definitions. Since these are objects, the first child is always the `{`.

Since `OBJBLOCK` is a token we create and is not defined in the source, it always has the same line/column as the first child. This AST is not sent the `log` and we don't look at the text/type of the AST in this check.

So for all purposes, the 2 are identical and results in no differences.